### ISSUE TITLE
So just to be clear, this isn't useable for TTS normalization?

### ISSUE THREAD

**Issue by xenotropic** [tomaarsen/TTSTextNormalization#1]
I'm looking for open source software that actually does English language TTS normalization. It seems like this code is not a candidate I should be looking at, because it requires a non-public dataset to be used? Did I read that right?

**Comment by tomaarsen** [comment]
Hello!

It is technically usable without the non-public dataset, but not particularly conveniently. I did not write this to be particularly "usable". However, with this script it is technically possible to run it using the [provided model](https://github.com/tomaarsen/TTSTextNormalization/blob/master/best/xgb_sub5750_5c_4w_5f_v1_model.dat):
```python

import pickle
import xgboost as xgb
import pandas as pd

from generate_submission import convert_data

xgb_model_path = r"best\xgb_sub5750_5c_4w_5f_v1_model.dat"

# Load the model
with open(xgb_model_path, "rb") as f:
    model = pickle.load(f)

test_sentence = "A baby giraffe is 13ft tall and weighs 150lb ."

sentence_series = pd.DataFrame(test_sentence.split())[0]
x_test = convert_data(sentence_series)

# Create a DMatrix to use in prediction
d_test = xgb.DMatrix(x_test)
# Predict classes using the model
preds = model.predict(d_test)

from converters.Plain      import Plain
from converters.Punct      import Punct
from converters.Date       import Date
from converters.Letters    import Letters
from converters.Cardinal   import Cardinal
from converters.Verbatim   import Verbatim
from converters.Decimal    import Decimal
from converters.Measure    import Measure
from converters.Money      import Money
from converters.Ordinal    import Ordinal
from converters.Time       import Time
from converters.Electronic import Electronic
from converters.Digit      import Digit
from converters.Fraction   import Fraction
from converters.Telephone  import Telephone
from converters.Address    import Address

labels = ["PLAIN", "PUNCT", "DATE", "LETTERS", "CARDINAL", "VERBATIM", "DECIMAL", "MEASURE", "MONEY", "ORDINAL", "TIME", "ELECTRONIC", "DIGIT", "FRACTION", "TELEPHONE", "ADDRESS"]
label_dict = {
    "PLAIN": Plain(),
    "PUNCT": Punct(),
    "DATE": Date(),
    "LETTERS": Letters(),
    "CARDINAL": Cardinal(),
    "VERBATIM": Verbatim(),
    "DECIMAL": Decimal(),
    "MEASURE": Measure(),
    "MONEY": Money(),
    "ORDINAL": Ordinal(),
    "TIME": Time(),
    "ELECTRONIC": Electronic(),
    "DIGIT": Digit(),
    "FRACTION": Fraction(),
    "TELEPHONE": Telephone(),
    "ADDRESS": Address(),
}

for before, pred in zip(sentence_series, preds):
    token_label = labels[int(pred)]
    label_converter = label_dict[token_label]
    print(f"{before:<10} => {label_converter.convert(before):<30} ({token_label})")
```
That said, the requirements are pretty strict. I had to install Python 3.6 again, and use the following requirements:
```
numpy==1.18.1
pandas==0.25.3
xgboost==0.90
scikit-learn
matplotlib
singleton_decorator
```

Beyond that, I think one of the dependency versions differed from what I originally used, as the performance seems to have degraded. The above script outputs:
```
A          => A                              (VERBATIM)
baby       => baby                           (PLAIN)
giraffe    => giraffe                        (PLAIN)
is         => is                             (PLAIN)
13ft       => 13ft                           (PLAIN)
tall       => tall                           (PLAIN)
and        => and                            (PLAIN)
weighs     => weighs                         (PLAIN)
150lb      => one hundred fifty pounds       (MEASURE)
.          => .                              (PUNCT)
```
For example `13ft` should have been classified as MEASURE, but it was classified as PLAIN instead.

In conclusion, I would recommend looking elsewhere for more recent and more usable programs. That said, I don't know of any.
Thank you for your interest, though!

- Tom Aarsen

**Comment by xenotropic** [comment]
Thank you so much for taking the time to write such a long and thoughtful reply. I'm someone new in looking at TTS, and I'm surprised at how far along a lot of TTS is (and how many options there are). But normalization, doesn't seem like it is so much. Anyway thanks again.
